examples/ui/egui/simple_ui.rs

```diff
@@ -0,0 +1,95 @@
+use nannou::prelude::*;
+use nannou_egui::{self, egui, Egui};
+
+fn main() {
+    nannou::app(model).update(update).run();
+}
+
+struct Settings {
+    resolution: u32,
+    scale: f32,
+    rotation: f32,
+    color: Srgb<u8>,
+    position: Vec2,
+}
+
+struct Model {
+    settings: Settings,
+    egui: Egui,
+}
+
+fn model(app: &App) -> Model {
+    // Create window
+    let window_id = app
+        .new_window()
+        .view(view)
+        .raw_event(raw_window_event)
+        .build()
+        .unwrap();
+    let window = app.window(window_id).unwrap();
+
+    let egui = Egui::from_window(&window);
+
+    Model {
+        egui,
+        settings: Settings {
+            resolution: 10,
+            scale: 200.0,
+            rotation: 0.0,
+            color: WHITE,
+            position: vec2(0.0, 0.0),
+        },
+    }
+}
+
+fn update(_app: &App, model: &mut Model, update: Update) {
+    let egui = &mut model.egui;
+    let settings = &mut model.settings;
+
+    egui.set_elapsed_time(update.since_start);
+    let ctx = egui.begin_frame();
+
+    egui::Window::new("Settings").show(&ctx, |ui| {
+        // Resolution slider
+        ui.add(egui::Label::new("Resolution:"));
+        ui.add(egui::Slider::new(&mut settings.resolution, 1..=40));
+
+        // Scale slider
+        ui.add(egui::Label::new("Scale:"));
+        ui.add(egui::Slider::new(&mut settings.scale, 0.0..=1000.0));
+
+        // Rotation slider
+        ui.add(egui::Label::new("Rotation:"));
+        ui.add(egui::Slider::new(&mut settings.rotation, 0.0..=360.0));
+
+        // Random color button
+        let clicked = ui.add(egui::Button::new("Random color")).clicked();
+
+        if clicked {
+            settings.color = rgb(random(), random(), random());
```

Comment:
Out of curiosity, does this work?
```suggestion
            settings.color = random();
```

Reply:
It doesn't, unfortunately, it gives this error:
![image](https://user-images.githubusercontent.com/33939820/173229120-21eecf60-3a07-4513-9bc8-4cfd5a207a12.png)